src/scaffold/entry_type/integrity.rs

```diff
@@ -846,54 +855,62 @@ fn add_entry_type_to_validation_arms(
-                                                                                entry_def.name.to_case(Case::Snake)
-                                                                            ).as_str()
-                                                                        )?;
-                                                                        entry_type_match
-                                                                            .arms
-                                                                            .push(new_arm);
-                                                                    }
+                                                if arm.body.as_ref().to_token_stream().to_string().contains("There are no entry types in this integrity zome") {
+                                                    *arm.body = syn::parse_str::<syn::Expr>(r#"{
+                                                        let original_action_hash = delete_entry.clone().action.deletes_address;
+                                                        let original_record = must_get_valid_record(original_action_hash)?;
+                                                        let original_action = original_record.action().clone();
+                                                        let original_action = EntryCreationAction::try_from(original_action)
+                                                            .map_err(|e| wasm_error!(e.to_string()))?;
+                                                        let app_entry_type = match original_action.entry_type() {
+                                                            EntryType::App(app_entry_type) => app_entry_type,
+                                                            _ => {
+                                                                return Ok(ValidateCallbackResult::Valid);
+                                                            }
+                                                        };
+                                                        let entry = match original_record.entry().as_option() {
+                                                            Some(entry) => entry,
+                                                            None => {
+                                                                if original_action.entry_type().visibility().is_public() {
+                                                                    return Ok(ValidateCallbackResult::Invalid(
+                                                                        "Original record for a delete of a public entry must contain an entry"
+                                                                            .to_string(),
+                                                                    ));
+                                                                } else {
+                                                                    return Ok(ValidateCallbackResult::Valid);
```

Comment:
I'm unsure about this one. I see that the previous scaffolding accepted all deletes other than `OpDelete::Entry` as valid, including for example private entries. But I'm wondering whether that's problematic? I don't know a lot about the workings of a private entry but I imagine that typically it should not be possible to delete a private entry of someone else.

Reply:
> I imagine that typically it should not be possible to delete a private entry of someone else

I suppose this would only be possible if they had a capability claim allowing for this, right? Regardless, I too feel like an invalid result should be returned from this.

I kept it this way since I noticed a similar check is made for `OpRecord::DeleteEntry`

Reply:
I have to say I'm still not clear on this one. I saw that you changed it to always rejecting deletion if it's a private entry but I'm not sure whether deleting your own private entry might in general an okay (and maybe desirable) thing to do. I also notice that scaffolding does not a way currently to scaffold private entries in the first place I think. So I'm not sure what to do here...

Reply:
Well...if scaffolding does not provide a way to scaffold private entries then maybe rejecting deletion of private entries by default is a way of forcing developers that change entry types to be private to add appropriate validation because their app won't work otherwise (and they hopefully notice this during testing).

Reply:
> I'm not sure whether deleting your own private entry might in general an okay (and maybe desirable) thing to do. 

This is my reasoning behind my change. 

> I also notice that scaffolding does not a way currently to scaffold private entries

Yes, perhaps having this configured via scaffolding an `entry-type` would guide on how to handle this case appropriately